Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -343,8 +365,8 @@ final class BuildPlanTests: XCTestCase {
             let llbuild = LLBuildManifestBuilder(plan, fileSystem: fs, observabilityScope: observability.topScope)
             try llbuild.generateManifest(at: yaml)
             let contents: String = try fs.readFileContents(yaml)
-            XCTAssertMatch(contents, .contains("""
-                    inputs: ["/Pkg/Sources/exe/main.swift","/path/to/build/release/PkgLib.swiftmodule"]
+            XCTAssertMatch(contents.replacingOccurrences(of: "\\\\", with: "\\"), .contains("""
```

Comment:
can you elaborate on why `replacingOccurrences(of: "\\\\", with: "\\")` is needed, seems like maybe hints at an issue?

Reply:
It isn't entirely so clear to me if this is a real issue or not.  What it comes down to is that the string interpolation will translate the `\` to `\\` as the path is treated a string and `\` is an escape sequence otherwise.  This replacement will undo that string conversion so that the match actually matches the string.  Now perhaps we should not emit the secondary `\` in the string conversion, but that seems like a difficult change to make (and would require evolution likely).

Reply:
@abertelrud wdyt?

Reply:
So if I understand this correctly, this is the YAML that gets sent to LLBuild.  And it needs any reverse solidus that represents a path separator in the YAML file to be escaped?  If so then that seems expected, though I think it would be clearer to:
1. have a fileprivate `escapedForYAML` property on String
2. use that on the right side of the `XCTAssertMatch` call
That makes it clearer, I think, that we're looking for the YAML representation of a path in this file.

Does that seem reasonable?

Reply:
I believe that is mostly the case (the only bit that I am uncertain about is the contents of the file).

As to the proposed solution, I agree with it in principle, not in practice.  I'd prefer `escaped` over `escapedForYAML` simply because we use YAML and JSON for various things and both of them will end up escaping the sequence and we will end up with another set of confusion.

Reply:
For the moment, I've added the extension to the file, but it is likely that we will need to hoist it as we make progress with the rest of the changes.